larpandoracontent/LArControlFlow/NeutrinoIdTool.h

```diff
@@ -24,13 +35,201 @@ class NeutrinoIdTool : public SliceIdBaseTool
      */
     NeutrinoIdTool();
 
-    void SelectOutputPfos(const SliceHypotheses &nuSliceHypotheses, const SliceHypotheses &crSliceHypotheses, pandora::PfoList &selectedPfos);
+    void SelectOutputPfos(const MasterAlgorithm *const pAlgorithm, const SliceHypotheses &nuSliceHypotheses, const SliceHypotheses &crSliceHypotheses, pandora::PfoList &selectedPfos);
 
 private:
+    class SliceFeatures;
+
+    typedef std::pair<unsigned int, float> UintFloatPair;
+    typedef std::vector<SliceFeatures> SliceFeaturesVector;
+
+    /**
+     *  @brief  Slice features class
+     */
+    class SliceFeatures
+    {
+    public:
+        /**
+         *  @brief  Constructor
+         *
+         *  @param  nuPfos input list of Pfos reconstructed under the neutrino hypothesis
+         *  @param  crPfos input list of Pfos reconstructed under the cosmic ray hypothesis
+         *  @param  pTool address of the tool using this class
+         */
+        SliceFeatures(const pandora::PfoList &nuPfos, const pandora::PfoList &crPfos, const NeutrinoIdTool *const pTool);
+
+        /**
+         *  @brief  Check if all features were calculable
+         */
```

Comment:
Ought to add e.g. `@return    boolean`, and provide details for all other functions returning anything other than void, although you will find this hasn't been done comprehensively throughout our entire code base.